builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go

```diff
@@ -332,6 +332,90 @@ func (uw *UserDeployerWhitelist) GetDeployedContracts(
 
+// SwapUserDeployer Allow swapping deployer keys in UserWhitelistContract
+func (uw *UserDeployerWhitelist) SwapUserDeployer(
+	ctx contract.Context, req *udwtypes.SwapUserDeployerRequest,
+) error {
+	if req.OldDeployerAddr == nil || req.NewDeployerAddr == nil {
+		return ErrInvalidRequest
+	}
+	if !ctx.FeatureEnabled(loomchain.UserDeployerWhitelistVersion1_2Feature, false) {
+		return errors.New("Feature not enabled")
+	}
+	dwAddr, err := ctx.Resolve("deployerwhitelist")
+	if err != nil {
+		return errors.Wrap(err, "unable to get address of deployer_whitelist")
+	}
+	if ctx.Has(deployerStateKey(loom.UnmarshalAddressPB(req.NewDeployerAddr))) {
+		return ErrDeployerAlreadyExists
+	}
+	// check if oldDeployerState exists and sender is the user who whitelisted the oldDeployerAddr,
+	// if so remove from userState and add the newDeployerAddr
+	oldDeployerAddr := loom.UnmarshalAddressPB(req.OldDeployerAddr)
+	var userDeployer UserDeployerState
+	err = ctx.Get(deployerStateKey(oldDeployerAddr), &userDeployer)
+	if err != nil {
+		return errors.Wrap(err, "no deployers found")
+	}
+	userAddr := ctx.Message().Sender
+	var userState UserState
+	if err := ctx.Get(userStateKey(userAddr), &userState); err != nil {
+		return errors.Wrap(err, "[UserDeployerWhitelist] Failed to load User State")
+	}
+	isInputDeployerAddrValid := false
+	survivedDeployers := make([]*types.Address, 0, len(userState.Deployers))
+	for _, addr := range userState.Deployers {
+		currentDeployerAddr := loom.UnmarshalAddressPB(addr)
+		if currentDeployerAddr.Compare(oldDeployerAddr) == 0 {
+			isInputDeployerAddrValid = true
+		} else {
+			survivedDeployers = append(survivedDeployers, addr)
+		}
+	}
+
+	if !isInputDeployerAddrValid {
+		return ErrDeployerDoesNotExist
+	}
+
+	survivedDeployers = append(survivedDeployers, req.NewDeployerAddr)
+	userState.Deployers = survivedDeployers
+	if err := ctx.Set(userStateKey(userAddr), &userState); err != nil {
+		return errors.Wrap(err, "failed to Save Deployers mapping in user state")
+	}
+	// Update DeployerWhitelist contract to reflect the deployer account swap
+	removeUserDeployerRequest := &RemoveUserDeployerRequest{
+		DeployerAddr: req.OldDeployerAddr,
+	}
+	if err := contract.CallMethod(ctx, dwAddr, "RemoveUserDeployer", removeUserDeployerRequest, nil); err != nil {
+		return errors.Wrap(err, "failed to remove deployer")
+	}
+	addUserDeployerRequest := &dwtypes.AddUserDeployerRequest{
+		DeployerAddr: req.NewDeployerAddr,
+	}
+	if err := contract.CallMethod(ctx, dwAddr, "AddUserDeployer", addUserDeployerRequest, nil); err != nil {
+		return errors.Wrap(err, "failed to whitelist deployer")
+	}
+
+	// Update deployerState in userDeployerWhitelist contract by adding new state corresponding to NewDeployerAddr with tierID same as oldDeployer and modifying older one corresponding to OldDeployerAddr
+	newDeployer := &UserDeployerState{
+		Address:   req.NewDeployerAddr,
+		TierID:    userDeployer.TierID,
+		Contracts: userDeployer.Contracts,
```

Comment:
I don't recall asking for this to change. The swap is supposed to be equivalent to removing the old key, and adding a new one, without paying the fee again. So the contracts deployed by the old key shouldn't be associated with the new key.